confmap/provider/httpsprovider/provider.go

```diff
@@ -0,0 +1,108 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package httpsprovider
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+
+	"go.opentelemetry.io/collector/confmap"
+	"go.opentelemetry.io/collector/confmap/provider/internal"
+)
+
+const (
+	schemeName = "https"
+)
+
+type httpsClient interface {
+	Get(url string) (resp *http.Response, err error)
+}
+
+type provider struct {
+	client http.Client
+}
+
+// New returns a new confmap.Provider that reads the configuration from a file.
+//
+// This Provider supports "https" scheme, and can be called with a "uri" that follows:
+//   https-uri : https://host/xxx
+//
+// One example for https-uri be like: https://localhost:4444/getConfig
+//
+// Examples:
+// `https://localhost:4444/getConfig` - (unix, windows)
+func New() confmap.Provider {
+	// create a certificate pool, then add the root CA into it
+	myCAPath := os.Getenv("SSL_CERT_FILE")
```

Comment:
This env var name is too generic and it is also a CA certificate, so maybe change to something like: `OTEL_HTTPS_CONFMAP_CA_CERT`?